### ISSUE TITLE
Handle long-press on TextField

### ISSUE THREAD

**Issue by deebov** [adobe/react-spectrum#695]
# 🙋 Feature Request

Handle long-press on TextField's steppers

## 🤔 Expected Behavior

1. User clicks and holds one of the steppers (increment, decrement) of TextField
2. Input value changes based on the action (inc. or dec.) once
3.  A delay
4. Input value starts to go up or do down continuously

## 😯 Current Behavior

1. User clicks and holds one of the steppers (increment, decrement) of TextField
2. Input value changes based on the action (inc. or dec.) once
3.  nothing happens

## 💁 Possible Solution

Listen to the press event. change the value based on the kind of action and set a timer at the same time. when the timer ends check if the press event is still active. if so then set an interval that changes the value. stop the interval then the press event is not active anymore

## 🔦 Context

I am building a design system builder. And I think Spectrum has the most suitable set of tools for my project.
many editors and number fields in real-applications have this feature. and it feels natural to the user. also, i can change the value only without using a keyboard.

## 💻 Examples

https://developer.microsoft.com/en-us/fluentui#/controls/web/spinbutton
if you check the first example you can see how it works.

P.s. also I have another couple of suggestions such as selecting the value on focus (see the example link) and validating the value on blur ( e.g. min set to 0; user enters -100; value set to the closest valid value (0) when the user moves the focus).

**Comment by snowystinger** [comment]
Are you using TextField with type=number?
I believe we're covering this `type` with a different component down the line, NumberInput.

**Comment by deebov** [comment]
Yes, I used the NumberField example from the storybook. I will take a look and try to implement this feature

**Comment by snowystinger** [comment]
Yeah, that component is not finished yet, it’s actually on hold because it needs a bit of a re-write. If you are still interested in contributing, the component would need to be finished which first requires finalizing the API and working out accessibility.
https://github.com/adobe-private/react-spectrum-v3/tree/master/specs/api
https://github.com/adobe-private/react-spectrum-v3/tree/master/specs/accessibility

**Comment by deebov** [comment]
@snowystinger would you emphasize what actually needs to be done before re-writing/editing the component? I will try to re-write the NumberField component with the features I mentioned above.

**Comment by deebov** [comment]
btw, how is an element `ref` handled in `@react-aria`? I can see that we _can_ pass a ref property to the final props objects. I think this would do the trick. But I'd like to know more if there's _the right way_ to access a `ref` from `@react-aria`

**Comment by snowystinger** [comment]
Well, for one, it needs entries in the specs for API and Accessibility. This component was started mostly as an experiment because we don't know its requirements yet and we wanted something a little more concrete to look at.

As for how to use ref's, the most similar one you probably have to look at is SearchField https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/searchfield/src/useSearchField.ts
Ref's should be passed from the component to the hook except in rare cases.
